Microsoft.Toolkit.Uwp.UI.Controls.Core/ImageEx/ImageExBase.Source.cs

```diff
@@ -24,9 +24,11 @@ public partial class ImageExBase
         /// </summary>
         public static readonly DependencyProperty SourceProperty = DependencyProperty.Register(nameof(Source), typeof(object), typeof(ImageExBase), new PropertyMetadata(null, SourceChanged));
 
-        private Uri _uri;
-        private bool _isHttpSource;
-        private CancellationTokenSource _tokenSource = null;
+        /// <summary>
+        /// Gets value tracking the currently requested source Uri. This can be helpful to use when implementing <see cref="AttachCachedResourceAsync(Uri)"/> where loading an image from a cache takes longer and the current container has been recycled and is no longer valid since a new image has been set.
+        /// </summary>
+        protected Uri CurrentSourceUri { get; private set; }
```

Comment:
I'm not 100% convinced about this property 🤔

As in, it seems to me that this is only mean to be used by inheriting classes overriding the loading method to check against the `Uri` they have as an argument, to detect whether the image has been changed while they were still loading. In that case wouldn't it be clearer and potentially less error prone to instead just keep a private field of type `CancellationTokenSource`, and pass a token to that virtual `LoadCachedImageAsync` method? That was people implementing that would just need to check whether or not the token is canceller, or they could just as easily just call [CancellationToken.ThrowIfCancellationRequested\(https://docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtoken.throwifcancellationrequested?view=net-5.0), since the corresponding exception is already property handled by the callee anyway. The control would just need to cancel the token whenever the image is changed, and then assign a new token source if needed and pass the token downstream.

Thoughts?

Reply:
Yup, sounds good.